transforms/universal/pdf2parquet/python/src/pdf2parquet_transform.py

```diff
@@ -0,0 +1,269 @@
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import io
+import uuid
+import zipfile
+import time
+from argparse import ArgumentParser, Namespace
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+from data_processing.utils.cli_utils import CLIArgumentProvider
+import filetype
+import pyarrow as pa
+from data_processing.transform import AbstractBinaryTransform, TransformConfiguration
+from data_processing.utils import TransformUtils, get_logger, str2bool
+from docling.datamodel.base_models import DocumentStream
+from docling.datamodel.document import ConvertedDocument, DocumentConversionInput
+from docling.document_converter import DocumentConverter
+from docling.pipeline.standard_model_pipeline import PipelineOptions
+
+
+logger = get_logger(__name__)
+
+shortname = "pdf2parquet"
+cli_prefix = f"{shortname}_"
+pdf2parquet_artifacts_path_key = f"artifacts_path"
+pdf2parquet_do_table_structure_key = f"do_table_structure"
+pdf2parquet_do_ocr_key = f"do_ocr"
+
+pdf2parquet_do_table_structure_default = True
+pdf2parquet_do_ocr_default = False
+
+pdf2parquet_artifacts_path_cli_param = f"{cli_prefix}{pdf2parquet_artifacts_path_key}"
+pdf2parquet_do_table_structure_cli_param = f"{cli_prefix}{pdf2parquet_do_table_structure_key}"
+pdf2parquet_do_ocr_cli_param = f"{cli_prefix}{pdf2parquet_do_ocr_key}"
+
+
+class Pdf2ParquetTransform(AbstractBinaryTransform):
+    """ """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, LangSelectorTransformRuntime.  If running inside the RayMutatingDriver,
+        these will be provided by that class with help from the RayMutatingDriver.
+        """
+
+        super().__init__(config)
+        
+        self.artifacts_path = config.get(pdf2parquet_artifacts_path_key, None)
+        if self.artifacts_path is not None:
+            self.artifacts_path = Path(self.artifacts_path)
+        self.do_table_structure = config.get(pdf2parquet_do_table_structure_key, pdf2parquet_do_table_structure_default)
+        self.do_ocr = config.get(pdf2parquet_do_ocr_key, pdf2parquet_do_ocr_default)
+
+        logger.info("Initializing models")
+        pipeline_options = PipelineOptions(
+            do_table_structure=self.do_table_structure,
+            do_ocr=self.do_ocr,
+        )
+        self._converter = DocumentConverter(artifacts_path=self.artifacts_path, pipeline_options=pipeline_options)
```

Comment:
Is the artifact_path a path in the local file system?  If so, you should also support S3 using a DataAccessFactory.